### PR TITLE
[Numpy] FFI: random.choice, take and clip

### PR DESCRIPTION
**Pull request by AntiZpvoh**
## Description ##
Add FFI support of 

- random.choice

- take

- clip

  | Old FFI (ctypes) (us) | New FFI (cython) (us)
-- | -- | --
random.choice | 70.73 | 23.31
take | 61.6 | 29.95
clip | 31.95 | 11.44

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/operator/numpy/random/np_choice_op.h [hunk 1]
```diff
@@ -53,6 +53,19 @@ struct NumpyChoiceParam : public dmlc::Parameter<NumpyChoiceParam> {
...
+    std::ostringstream a_s, size_s, replace_s, weighted_s;
+    a_s << a;
+    size_s << size;
+    replace_s << replace;
+    weighted_s << weighted;
+    (*dict)["a"] = a_s.str();
+    (*dict)["size"] = size_s.str();
+    (*dict)["replace"] = replace_s.str();
+    (*dict)["weighted"] = weighted_s.str();
+    // We do not set ctx, because ctx has been set in dict instead of InitOpParam.
```
Comment (haojin2) [inline src/operator/numpy/random/np_choice_op.h]: get rid of this comment, looks like it does not apply to this op.

**Comment by haojin2** [comment]
please also fix the sanity issues.

File: src/api/operator/tensor/indexing_op.cc [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+namespace mxnet {
+
+MXNET_REGISTER_API("_npi.take")
+.set_body([](runtime::MXNetArgs args, runtime::MXNetRetValue* ret) {
+  using namespace runtime;
+  const nnvm::Op* op = Op::Get("_npi_take");
+  nnvm::NodeAttrs attrs;
+  op::TakeParam param;
+  NDArray** inputs = new NDArray*[2]();
```
Comment (hzfan) [inline src/api/operator/tensor/indexing_op.cc]: Please avoid new here. Use `NDArray* inputs[2]`.

File: src/api/operator/tensor/matrix_op.cc [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+namespace mxnet {
+
+MXNET_REGISTER_API("_npi.clip")
+.set_body([](runtime::MXNetArgs args, runtime::MXNetRetValue* ret) {
+  using namespace runtime;
+  const nnvm::Op* op = Op::Get("_npi_clip");
+  nnvm::NodeAttrs attrs;
+  op::ClipParam param;
+  NDArray** inputs = new NDArray*[1]();
```
Comment (hzfan) [inline src/api/operator/tensor/matrix_op.cc]: Same, please avoid new. Use `NDArray* inputs[1]`.

File: src/api/operator/numpy/random/np_choice_op.cc [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+namespace mxnet {
+
+MXNET_REGISTER_API("_npi.choice")
+.set_body([](runtime::MXNetArgs args, runtime::MXNetRetValue* ret) {
+  using namespace runtime;
+  const nnvm::Op* op = Op::Get("_npi_choice");
+  nnvm::NodeAttrs attrs;
+  op::NumpyChoiceParam param;
+
+  NDArray** in = new NDArray*[2];
```
Comment (hzfan) [inline src/api/operator/numpy/random/np_choice_op.cc]: Same, please avoid new

**Comment by AntiZpvoh** [comment]
@mxnet-bot run ci [ci/jenkins/mxnet-validation/centos-cpu, ci/jenkins/mxnet-validation/unix-gpu]

**Comment by AntiZpvoh** [comment]
@mxnet-bot run ci [centos-cpu, unix-gpu]

**Comment by AntiZpvoh** [comment]
@mxnet-bot run ci [unix-gpu]

**Pull request merged**